subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/matcher/OverlapMatcher.kt

```diff
@@ -0,0 +1,52 @@
+package com.avito.android.test.matcher
+
+import android.graphics.Rect
+import android.view.View
+import android.view.ViewGroup
+import org.hamcrest.Description
+import org.hamcrest.TypeSafeMatcher
+
+/**
+ * Check if there is an intersection of the target view's Rect and Rect of any view with higher z-order.
+ *
+ * How it works:
+ * 1. It checks siblings of target view that are below in child list of the parent
+ * 1. Check parent siblings that are below in child list of parent's parent
+ * 1. Check parent's parent siblings... and so on recursively
+ */
+class OverlapMatcher : TypeSafeMatcher<View>() {
+
+    override fun describeTo(description: Description) {
+        description.appendText("overlapped by another views on the same screen")
+    }
+
+    override fun matchesSafely(view: View): Boolean {
+        val viewRect = Rect()
+        view.getGlobalVisibleRect(viewRect)
+
+        return view.parentViewGroup?.iterateViewsBelow(view) { viewBelow ->
```

Comment:
I consider we need to use `rootView`

Reply:
Why do we need to use `rootView`? This method should iterate views from current parent up to root view. Check the docs above the class.